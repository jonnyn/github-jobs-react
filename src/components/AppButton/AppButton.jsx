import React from 'react';
import { Button, Spinner } from 'react-bootstrap';
import PropTypes from 'prop-types';

const AppButton = ({label, isLoading}) => {
    if (isLoading) {
        return (
            <Button variant="primary" disabled>
                <Spinner
                    as="span"
                    animation="grow"
                    size="sm"
                    role="status"
                    aria-hidden="true"
                />
                Loading...
            </Button>
        );
    }

    return (
        <Button variant="primary" type="submit" block>{label}</Button>
    );
}

AppButton.propTypes = {
    label: PropTypes.string.isRequired,
    isLoading: PropTypes.bool
 };

export default AppButton;