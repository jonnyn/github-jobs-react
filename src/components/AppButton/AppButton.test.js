import React from 'react';
import AppButton from './AppButton';
import renderer from 'react-test-renderer';

it('renders Go button when is not loading', () => {
  const component = renderer.create(<AppButton label='Go' />,);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders loading button when isLoading set to true', () => {
    const component = renderer.create(<AppButton label='Go' isLoading />,);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });