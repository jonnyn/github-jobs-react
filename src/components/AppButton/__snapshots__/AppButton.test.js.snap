// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Go button when is not loading 1`] = `
<button
  className="btn btn-primary btn-block"
  disabled={false}
  type="submit"
>
  Go
</button>
`;

exports[`renders loading button when isLoading set to true 1`] = `
<button
  className="btn btn-primary"
  disabled={true}
  type="button"
>
  <span
    aria-hidden="true"
    className="spinner-grow spinner-grow-sm"
    role="status"
  />
  Loading...
</button>
`;
